lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -291,13 +296,108 @@ MHWRender::MSamplerStateDesc _GetSamplerStateDesc(const HdMaterialNode& node)
     return desc;
 }
 
+MHWRender::MTexture* _LoadUdimTexture(
+    const std::string& path, bool& isColorSpaceSRGB, MFloatArray& uvScaleOffset)
+{
+#if USD_VERSION_NUM >= 2002
+    /*
+        For this method to work path needs to be an absolute file path, not an asset path.
+        That means that this function depends on the changes in 4e426565 to materialAdapther.cpp
+        to work. As of my writing this 4e426565 is not in the USD that MayaUSD normally builds
+        against so this code will fail, because UsdImaging_GetUdimTiles won't file the tiles
+        because we don't know where on disk to look for them.
+
+        https://github.com/PixarAnimationStudios/USD/commit/4e42656543f4e3a313ce31a81c27477d4dcb64b9
+    */
+
+    // test for a UDIM texture
+    if (!GlfIsSupportedUdimTexture(path))
+        return nullptr;
+    
+    /*
+        Maya's tiled texture support is implemented quite differently from Usd's UDIM support.
+        In Maya the texture tiles get combined into a single big texture, downscaling each tile
+        if necessary, and filling in empty regions of a non-square tile with the undefined color.
+
+        In USD the UDIM textures are stored in a texture array that the shader uses to draw.
+    */
+
+    MHWRender::MRenderer* const renderer = MHWRender::MRenderer::theRenderer();
+    MHWRender::MTextureManager* const textureMgr =
+        renderer ? renderer->getTextureManager() : nullptr;
+    if (!TF_VERIFY(textureMgr)) {
+        return nullptr;
+    }
+
+    MString textureName(path.c_str()); // used for caching, using the string with <UDIM> in it is fine
+    MStringArray tilePaths;
+    MFloatArray tilePositions;
+    // HdSt sets the tile limit to the max number of textures in an array of 2d textures. We don't
+    // want to set it too high because UsdImaging_GetUdimTiles will search for every one on disk.
+    int tileLimit = 100;
+    std::vector<std::tuple<int, TfToken>> tiles = UsdImaging_GetUdimTiles(path, tileLimit);
```

Comment:
Maybe we need to add a TF_WARN if there are more tiles than 100?

Reply:
UsdImaging_GetUdimTiles looks for 100 tiles on disk, so I don't directly know if there is a 101st tile to warn about. What I will do is set the tile limit to 100, but ask for 101 tiles. If the 101st tile is present I will emit the warning.

Reply:
I didn't know it is limit from USD side. If so, maybe rename the tileLimit to usdUdimTileLimit to be more specific?